controllers/metallb_controller.go

```diff
@@ -68,8 +70,27 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
-	err = r.syncMetalLBResources(instance)
+	isProgressing, err := r.syncMetalLBResources(instance)
 	if err != nil {
+		if err := status.Update(context.TODO(), r.Client, instance, status.ConditionDegraded, "FailedToSyncMetalLBResources", err.Error()); err != nil {
+			logger.Error(err, "Failed to update metallb status", "Desired status", status.ConditionDegraded)
+		}
+		return ctrl.Result{}, err
+	}
+	if isProgressing {
```

Comment:
Not sure what is the right way, but I feel you can't have progressing and available at the same time.
Can you refactor the status calculation and have a single status.Update call here?

Reply:
Removed the isProgressing flag. We will only update to Progressing when we detect that the deployment and ds are not ready yet. Should any error on the way occur, we'll update to degraded.